File: lab3DemchukKI-301/src/KI301/Demchuk/lab3/Truck.java
package KI301.Demchuk.lab3;

/**
 * Клас `Truck` представляє об'єкт вантажної машини, яка є підкласом класу `Car`.
 * Реалізує інтерфейси `CargoTransport` та `Trailer`.
 */
public class Truck extends KI301.Demchuk.lab3.Car implements CargoTransport, Trailer {
    private double cargoWeight;
    private boolean hasTrailer;

    /**
     * Завантажує вантаж на вантажну машину.
     *
     * @param weight Вага завантаженого вантажу (у кілограмах).
     */
    @Override
    public void loadCargo(double weight) {
        cargoWeight += weight;
    }

    /**
     * Розвантажує вантаж з вантажної машини.
     */
    @Override
    public void unloadCargo() {
        cargoWeight = 0;
    }

    /**
     * Повертає вагу завантаженого вантажу на вантажній машині.
     *
     * @return Вага завантаженого вантажу (у кілограмах).
     */
    @Override
    public double getCargoWeight() {
        return cargoWeight;
    }

    /**
     * Приєднує причіп до вантажної машини.
     */
    @Override
    public void attachTrailer() {
        hasTrailer = true;
    }

    /**
     * Від'єднує причіп від вантажної машини.
     */
    @Override
    public void detachTrailer() {
        hasTrailer = false;
    }

    /**
     * Перевіряє, чи причіп приєднаний до вантажної машини.
     *
     * @return True, якщо причіп приєднаний, false - у протилежному випадку.
     */
    @Override
    public boolean hasTrailer() {
        return hasTrailer;
    }
}

File: Lab9DemchukKI-301/Vencile/truck.py
from Vencile.car import Car

class Truck(Car):
    def __init__(self, brand, model, color, mileage=0, tank_capacity=0, fuel_consumption=0):
        """
        Initialize a Truck object, inheriting from the Car class.

        Args:
            brand (str): The brand of the truck.
            model (str): The model of the truck.
            color (str): The color of the truck.
            mileage (float): The current mileage of the truck (default is 0).
            tank_capacity (float): The tank capacity of the truck (default is 0).
            fuel_consumption (float): The fuel consumption of the truck (default is 0).
        """
        super().__init__(brand, model, color, mileage, tank_capacity, fuel_consumption)
        self.cargo_weight = 0
        self.has_trailer = False

    def load_cargo(self, weight):
        """
        Load cargo onto the truck.

        Args:
            weight (float): The weight of the cargo in kilograms.
        """
        self.cargo_weight += weight

    def unload_cargo(self):
        """
        Unload cargo from the truck.
        """
        self.cargo_weight = 0

    def get_cargo_weight(self):
        """
        Get the weight of the loaded cargo on the truck.

        Returns:
            float: The weight of the loaded cargo in kilograms.
        """
        return self.cargo_weight

    def attach_trailer(self):
        """
        Attach a trailer to the truck.
        """
        self.has_trailer = True

    def detach_trailer(self):
        """
        Detach the trailer from the truck.
        """
        self.has_trailer = False

    def truck_has_trailer(self):
        """
        Check if the truck has a trailer attached.

        Returns:
            bool: True if the truck has a trailer, False otherwise.
        """
        return self.has_trailer
